File: src/main/java/dev/robothanzo/werewolf/security/SessionRepository.java
package dev.robothanzo.werewolf.security;

import dev.robothanzo.werewolf.database.documents.Session;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface SessionRepository extends MongoRepository<Session, String> {
    Optional<Session> findByGuildId(long guildId);

    void deleteByGuildId(long guildId);
}

File: src/main/kotlin/dev/robothanzo/werewolf/security/SessionRepository.kt
package dev.robothanzo.werewolf.security

import dev.robothanzo.werewolf.database.documents.Session
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface SessionRepository : MongoRepository<Session, String> {
    fun findByGuildId(guildId: Long): Optional<Session>
    fun deleteByGuildId(guildId: Long)
}
